test/ReactiveSocketTest.cpp

```diff
@@ -311,4 +319,9 @@ TEST(ReactiveSocketTest, Destructor) {
   for (size_t i = 0; i < 2; ++i) {
     clientSock->requestSubscription(originalPayload->clone(), clientInputs[i]);
   }
+
+  //  EXPECT_CALL(clientStats, socketClosed_());
```

Comment:
I couldn't get the mocks behaving nicely.  It's called on destructor but it seems like the mocks run assertions on shutdown, so I was getting errors both ways. :(